_posts/2021-01-19-bazel-4-0.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: posts
+title: "Bazel 4.0"
+authors:
+  - communikit
+---
+
+Bazel 4.0 is a major release and Bazel’s first LTS release. It contains new features and backwards incompatible changes.
+
+## New features
+
+* Starlark-defined flags can now be shorthanded using `--flag_alias`.
+* Support has been added for [building with a Java 15 toolchain](https://github.com/bazelbuild/bazel/pull/12168).
+* Targets incompatible with the platform [can now be skipped](https://github.com/bazelbuild/bazel/issues/3780). See the [Platforms](https://docs.bazel.build/versions/master/platforms.html) documentation for more details.
+* Java workers are now [multithreaded by default](https://github.com/bazelbuild/bazel/issues/12165), reducing memory overhead.
+* Bazel 4.0 is our first release with [long-term support](https://blog.bazel.build/2020/11/10/long-term-support-release.html).
+
+## Incompatible changes
+
+The following flags have been flipped:
+
+* [`--incompatible_blacklisted_protos_requires_proto_info`](https://github.com/bazelbuild/bazel/issues/11694)
+* [`--incompatible_force_strict_header_check_from_starlark`](https://github.com/bazelbuild/bazel/issues/11679)
+* [`incompatible_require_linker_input_cc_api`](https://github.com/bazelbuild/bazel/issues/10860): C++ Starlark API now requires `linker_inputs` wrapping `library_to_link`.
+* [`--incompatible_proto_output_v2`](https://github.com/bazelbuild/bazel/issues/10358) for aquery, which applies a new, more compact aquery proto output format.
+
+The following flag has been added:
+
+* [`--incompatible_display_source_file_location`](https://github.com/bazelbuild/bazel/issues/8900) has been added for `bazel query location=output` to print the line 1 location of the actual source files instead of the source file targets.
+
+The following flags have been enabled by default:
```

Comment:
As per https://bazel.build/basics/release-notes.html#guidelines: "Do NOT make any promises about future functionality. Avoid “this flag will be removed” or “this will be changed.” It introduces uncertainty. The first thing the user will wonder is “when?” and we don’t want them to start worrying about their current builds breaking at some unknown time."

Is the note still warranted? Also, I am merging this section with the "flipped" flags section, since we are defining all of them as being enabled by default.

Reply:
As per https://bazel.build/basics/release-notes.html#guidelines : Do NOT make any promises about future functionality. Avoid “this flag will be removed” or “this will be changed.” It introduces uncertainty. The first thing the user will wonder is “when?” and we don’t want them to start worrying about their current builds breaking at some unknown time.

would this note still be warranted? Also, these flags have been merged with the section detailing "flipped" flags.